tiledb/CMakeLists.txt

```diff
@@ -187,37 +187,28 @@ if (TILEDB_SERIALIZATION)
     ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/rest/curl.cc
   )
 
-  #NOTE: These properties are being set possibly *before* tiledb-rest.capnp.c++ has been generated.
   if(NOT WIN32)
-    if($false)  #section will be needed when mov *n*x to v0.8.0
-    #capnproto v0.8.0 requires at least c++14, msvc2017 seems ok, gcc 7.5 is not by default
     set_source_files_properties(
       ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/rest/rest_client.cc
       ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/array_schema.cc
+      ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/config.cc
       ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/query.cc
       ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/tiledb-rest.capnp.c++
       PROPERTIES
-      #eventually, not yet in cmake 3.3, added in 3.11 - COMPILE_OPTIONS "-std=c++14"
-      #   C4267: 'argument': conversion from 'size_t' to '<various-lesser-types>', possible loss of data
-      #   C4100 '<some-name>' unreferenced format parameter
-        COMPILE_FLAGS "-std=c++14"
+        COMPILE_FLAGS "-Wno-unused-parameter"
     )
-    endif()
   endif()
   if(MSVC)
     set_source_files_properties(
       ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/rest/rest_client.cc
       ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/array_schema.cc
+      ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/config.cc
       ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/query.cc
       ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/tiledb-rest.capnp.c++
```

Comment:
tiledb/sm/c_api/tiledb.cc is also building with warnings (9 instances of C4100) for me with vs2017, should it be added here (possibly in 'NOT WIN32' branch above as well)?